src/transformers/models/pix2struct/modeling_pix2struct.py

```diff
@@ -1823,20 +1777,26 @@ def generate(
                         ],
                         dim=-1,
                     )
-        elif decoder_input_ids is None:
-            decoder_input_ids = (
-                torch.LongTensor([[self.decoder_start_token_id]]).repeat(batch_size, 1).to(image_embeds.device)
-            )
+        elif input_ids is None:
```

Comment:
Similarly to the comment above, if `config.is_encoder_decoder` was set to `True`, we shouldn't need these lines.

Let's add a comment in case we can't retroactively fix it

Reply:
Ok, we can still update the checkpoints on the hub to have `config.is_encoder_decoder=True`

Reply:
Update: no need to update any checkpoints on the hub